File: src/main/java/de/com/fdm/tmi/TwitchMessageDto.java
package de.com.fdm.tmi;

import com.github.twitch4j.chat.events.channel.ChannelMessageEvent;

public record TwitchMessageDto(String channel, String message) {
    public static TwitchMessageDto fromEvent(ChannelMessageEvent event) {
        return new TwitchMessageDto(event.getChannel().getName(), event.getMessage());
    }
}

File: src/main/java/de/com/fdm/tmi/TwitchMessageDto.kt
package de.com.fdm.tmi

import com.github.twitch4j.chat.events.channel.ChannelMessageEvent

data class TwitchMessageDto(
    val channel: String,
    val message: String,
    val userId: String,
    val rawMsg: String,
) {
    constructor(event: ChannelMessageEvent) : this(
        event.channel.name,
        event.message,
        event.user.id,
        event.messageEvent.rawMessage
    )
}
